Source/VFECore/Genes/Harmony/PawnRenderer_GetBodyOverlayMeshSet.cs

```diff
@@ -56,10 +56,10 @@ public static class PawnRenderer_BaseHeadOffsetAt_Patch
 
         static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
         {
-            var headOffset = AccessTools.Field("BodyTypeDef:headOffset");
-            var pawn = AccessTools.Field("PawnRenderer:pawn");
-            var bodyScaleFactor = AccessTools.Method("PawnRenderer_BaseHeadOffsetAt_Patch:LifeStageFactorUpdated");
-            var codes = instructions.ToList();
+            FieldInfo headOffset = AccessTools.Field(typeof(BodyTypeDef), nameof(BodyTypeDef.headOffset));
+            FieldInfo pawn = AccessTools.Field(typeof(PawnRenderer), "pawn");
+            MethodInfo bodyScaleFactor = ((Func<Vector2, Pawn, Vector2>)LifeStageFactorUpdated).Method;
```

Comment:
```suggestion
            var bodyScaleFactor = AccessTools.Method(typeof(PawnRenderer_BaseHeadOffsetAt_Patch), nameof(LifeStageFactorUpdated);
```
I don't see why you did the way with the delegate. It looks rather confusing at first glance, and makes the code less readable.

Reply:
This is the change that actually fixed the error for me.
I have tested your edit (after fixing the missing parenthesis), and AccessTools fails to grab that patch by name (my guess is due to that method is defined below the AccessTools call, but I haven't verified this to be the real cause).

For methods within our control (defined in our namespace), using a compile-time delegate has been more reliable to me than letting AccessTools resolve a name string at runtime to get our own method (this case being an example). 

Additionally, a delegate should be faster since AccessTools has to [search through every method under the type](https://github.com/pardeike/Harmony/blob/master/Harmony/Tools/AccessTools.cs/#L413).

The drawback of using a delegate is mostly that you have to explicitly specify the method (which like you said may hurt readability), and I use AccessTools for Methods/Fields that I can't control -- namely Vanilla RW codes whose method signature changes often.

Reply:
Ah, found the issue. The actual reason it doesn’t work is because the class is wrong. It should be:
```suggestion
            var bodyScaleFactor = AccessTools.Method(typeof(PawnRenderer_GetBodyOverlayMeshSet ), nameof(LifeStageFactorUpdated));
```
And AccessTools actually searches all the supertypes, not all the methods. Regardless, this only happens once, during load, which means speed isn’t all that important here.

Reply:
Ahh that make sense.
I am merging your suggestion if you wish since then this (Delegate vs AccessTools) is IMO a choice of style.
Though you might need to resubmit a suggestion since I just had a commit that changed it to var a few mins ago.

Reply:
Hmm yeah you're right. I think the reason the string method didn't work is because it's a nested type. But using `typeof` and `AccessTools` should work.

Reply:
I just realized that I couldn't replicate the error with just VEF, did some more testing, and found the real root cause:
The mod [Appearance clothes](https://steamcommunity.com/sharedfiles/filedetails/?id=1537867047) has a patch class also named "PawnRenderer_BaseHeadOffsetAt_Patch", this is why the original code (with string typeColonName)'s AccessTools call failed.
However, in such case, using typeof and AccessTools does resolve the problem.
And I somehow failed to replicate the error I believe I saw earlier with typeof and AccessTools.

While this is an unsatisfying answer for me, it shows that you are correct that "using typeof and AccessTools should work".
I am going to modify the PR according to your suggestion so you can merge it. If I later successfully replicate the problem I had earlier with using typeof and AccessTools, I will start another PR and explain how to replicate that.